src/data/general-cs-quiz.ts

```diff
@@ -1677,8 +1677,43 @@ const generalCSQuiz = [
   Distractor3: "It only accepts elements of the same type.",
   Explanation: "A priority queue is essential in scenarios where certain tasks need to be handled in order of importance or urgency.",
   Link: "https://www.freecodecamp.org/news/priority-queue-implementation-in-java/"
+},
+{
+  Question: "What is a binary search algorithm?",
+  Answer: "Binary search efficiently finds values in sorted arrays by dividing the search space in half until the target is found or determined to be absent.",
+  Distractor1: "Binary search is used for searching unsorted arrays.",
+  Distractor2: "Binary search only works with non-integer values.",
+  Distractor3: "Binary search has a time complexity of O(n).",
+  Explanation: "Binary search is a fundamental and highly efficient algorithm for searching in sorted data structures like arrays and is based on the principle of divide and conquer.",
+  Link: "https://en.wikipedia.org/wiki/Binary_search_algorithm",
```

Comment:
always use a freeCodeCamp resource first since that is what we are trying to promote.
Then move to wikipedia if no freeCodeCamp resource is available or applicable 

```suggestion
  Link: "https://www.freecodecamp.org/news/binary-search-algorithm-and-time-complexity-explained/",
```